ionoscloud/resource_nfs_cluster_test.go

```diff
@@ -0,0 +1,219 @@
+//go:build all || nfs || nfs_cluster
+
+package ionoscloud
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-testing/helper/resource"
+	"github.com/hashicorp/terraform-plugin-testing/terraform"
+	"github.com/ionos-cloud/terraform-provider-ionoscloud/v6/services"
+	"github.com/ionos-cloud/terraform-provider-ionoscloud/v6/utils/constant"
+)
+
+func TestAccNFSCluster_basic(t *testing.T) {
+	resource.Test(t, resource.TestCase{
+		PreCheck: func() {
+			testAccPreCheck(t)
+		},
+		ExternalProviders: map[string]resource.ExternalProvider{
+			"random": {
+				VersionConstraint: "3.4.3",
+				Source:            "hashicorp/random",
+			},
+			"time": {
+				Source:            "hashicorp/time",
+				VersionConstraint: "0.11.1",
+			},
+		},
+		ProviderFactories: testAccProviderFactories,
+		CheckDestroy:      testAccCheckNFSClusterDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccCheckNFSClusterConfig_basic,
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckNFSClusterExists("ionoscloud_nfs_cluster.example"),
+					resource.TestCheckResourceAttr("ionoscloud_nfs_cluster.example", "name", "example"),
+					resource.TestCheckResourceAttr("ionoscloud_nfs_cluster.example", "location", "de/txl"),
+					resource.TestCheckResourceAttr("ionoscloud_nfs_cluster.example", "size", "2"),
+					resource.TestCheckResourceAttr("ionoscloud_nfs_cluster.example", "nfs.0.min_version", "4.2"),
+				),
+			},
+			{
+				Config: testAccCheckNFSClusterConfig_update,
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckNFSClusterExists("ionoscloud_nfs_cluster.example"),
+					resource.TestCheckResourceAttr("ionoscloud_nfs_cluster.example", "name", "example_updated"),
+					resource.TestCheckResourceAttr("ionoscloud_nfs_cluster.example", "location", "de/txl"),
+					resource.TestCheckResourceAttr("ionoscloud_nfs_cluster.example", "size", "2"),
+					resource.TestCheckResourceAttr("ionoscloud_nfs_cluster.example", "nfs.0.min_version", "4.2"),
+				),
+			},
+			{
+				Config: testAccDataSourceNFSClusterMatchName,
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttrPair("data.ionoscloud_nfs_cluster.data_with_name", "name", "ionoscloud_nfs_cluster.example", "name"),
```

Comment:
@cristiGuranIonos consider also adding checks for other fields since we want to see if those are set in the data source accordingly.

Reply:
more complex tests TBD after resources are available